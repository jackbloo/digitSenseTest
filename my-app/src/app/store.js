import { configureStore } from '@reduxjs/toolkit';
import dataReducer from '../reducer/data/reducer';

export const store = configureStore({
  reducer: {
    dataPicker: dataReducer
  },
});
