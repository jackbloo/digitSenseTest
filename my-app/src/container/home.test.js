import React from 'react';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux';
import { store } from '../app/store';
import Home from './home'
import {Router} from 'react-router-dom'
import {createMemoryHistory} from 'history'


describe('Unit Test for Home', () => {
    const history=createMemoryHistory()
    test('Should be in the document', () => {
      render(
          <Provider store={store} >
              <Router navigator={history} location={history.location}>
            <Home />
            </Router>
          </Provider>
        );
      const card = screen.getByTestId('home-container')
      expect(card).toBeInTheDocument();
      });

      test('Should match the snapshot',  () => {
         render(
             <Provider store={store} >
              <Router navigator={history} location={history.location}>
            <Home />
            </Router>
             </Provider>
           );
         const card = screen.getByTestId('home-container')
         expect(card).toMatchSnapshot()
         });
})

