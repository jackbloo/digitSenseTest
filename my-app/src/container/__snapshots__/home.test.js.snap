// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test for Home Should match the snapshot 1`] = `
<div
  class="sc-furwcr jThAww"
  data-testid="home-container"
>
  <div
    class="sc-pVTFL eRjRBi"
  >
    <div
      class="sc-kDTinF yYeNI"
    >
      DigitSense Exam
    </div>
    <div
      class="sc-iqseJM bqyZiN"
    >
      New Post
      <div
        class="sc-crHmcD bVzDQc fa fa-plus icon"
      />
    </div>
    <div
      class="sc-kDTinF yYeNI"
    >
      Post
    </div>
    <div
      class="sc-jrQzAO hZhYgW"
    >
      <div
        class="sc-egiyK iAnnCa"
      />
    </div>
  </div>
</div>
`;

exports[`Unit Test for NewPost Should match the snapshot 1`] = `
<div
  class="sc-furwcr jThAww"
  data-testid="home-container"
>
  <div
    class="sc-pVTFL eRjRBi"
  >
    <div
      class="sc-kDTinF yYeNI"
    >
      DigitSense Exam
    </div>
    <div
      class="sc-iqseJM bqyZiN"
    >
      New Post
      <div
        class="sc-crHmcD bVzDQc fa fa-plus icon"
      />
    </div>
    <div
      class="sc-kDTinF yYeNI"
    >
      Post
    </div>
    <div
      class="sc-jrQzAO hZhYgW"
    >
      <div
        class="sc-egiyK iAnnCa"
      />
    </div>
  </div>
</div>
`;
